ios/Classes/PTFlutterDocumentController.m

```diff
@@ -0,0 +1,728 @@
+        } error:&error];
+        
+        if (!error) {
+            return [fdfDoc SaveAsXFDFToString];
+        }
+        NSLog(@"Error: %@", error.description);
+    }
+    return Nil;
+}
+
+-(NSString*)generateAnnotationWithActionString:(PTAnnot *)annotation onPageNumber:(unsigned long)pageNumber action:(NSString *)action
+{
+    
+    if (annotation.IsValid) {
+        
+        __block NSString *uniqueId;
+        
+        NSError *error;
+        [self.pdfViewCtrl DocLockReadWithBlock:^(PTPDFDoc * _Nullable doc) {
+            PTObj *uniqueIdObj = [annotation GetUniqueID];
+            if ([uniqueIdObj IsValid] && [uniqueIdObj IsString]) {
+                uniqueId = [uniqueIdObj GetAsPDFText];
+            }
+        } error:&error];
+        
+        if (error) {
+            NSLog(@"An error occurred: %@", error);
+            return nil;
+        }
+        
+        NSDictionary *annotDict = @{
+            PTAnnotationIdKey: uniqueId,
+            PTAnnotationPageNumberKey: [NSNumber numberWithLong:pageNumber],
+        };
+        
+        NSDictionary *resultDict = @{
+            PTAnnotationListKey: @[annotDict],
+            PTActionKey: action,
+        };
+        
+        return [PdftronFlutterPlugin PT_idToJSONString:resultDict];
+    }
+    return nil;
+}
+
+#pragma mark - <PTPDFViewCtrlDelegate>
+
+- (void)pdfViewCtrl:(PTPDFViewCtrl *)pdfViewCtrl onSetDoc:(PTPDFDoc *)doc
+{
+    [super pdfViewCtrl:pdfViewCtrl onSetDoc:doc];
+
+    if (self.local && !self.documentLoaded) {
+        self.needsDocumentLoaded = YES;
+    }
+    else if (!self.local && !self.documentLoaded && self.needsRemoteDocumentLoaded) {
+        self.needsDocumentLoaded = YES;
+    }
+    else if (!self.local && !self.documentLoaded && self.coordinatedDocument.fileURL) {
+        self.needsDocumentLoaded = YES;
+    }
+}
+
+- (void)pdfViewCtrl:(PTPDFViewCtrl *)pdfViewCtrl downloadEventType:(PTDownloadedType)type pageNumber:(int)pageNum message:(NSString *)message
+{
+    if (type == e_ptdownloadedtype_finished && !self.documentLoaded) {
+        self.needsRemoteDocumentLoaded = YES;
+    }
+
+    [super pdfViewCtrl:pdfViewCtrl downloadEventType:type pageNumber:pageNum message:message];
+}
+
+#pragma mark - <PTOutlineViewControllerDelegate>
+
+- (void)outlineViewControllerDidCancel:(PTOutlineViewController *)outlineViewController
+{
+    [outlineViewController dismissViewControllerAnimated:YES completion:nil];
+}
+
+#pragma mark - <PTAnnotationViewControllerDelegate>
+
+- (void)annotationViewControllerDidCancel:(PTAnnotationViewController *)annotationViewController
+{
+    [annotationViewController dismissViewControllerAnimated:YES completion:nil];
+}
+
+#pragma mark - Viewer Settings
+
+- (void)initViewerSettings
+{
+    _hideAnnotationToolbarSwitcher = NO;
+    _hideTopToolbars = NO;
+    _hideTopAppNavBar = NO;
+    _readOnly = NO;
+    
+    _showNavButton = YES;
+}
+
+- (void)applyViewerSettings
```

Comment:
This method is a bit confusing to me, how it is applying some settings based on properties, but others are hard coded. Are hard-coded defaults used simply because of when this method is called? Can the user not change these properties and then later have this method override them? I think this method needs a new name, or its logic needs changing.